### DISCUSSION TITLE
Full syntax grammar for Xonsh

### DISCUSSION THREAD

**Discussion by tomodachi94** [xonsh/xonsh#4925] [Q&A] [answered]
Hey all,

I'm building custom syntax highlighting for [an indie editor](https://micro-editor.github.io), and I was wondering if there was a full grammar for Xonsh syntax which includes Xonsh-exclusive primitives and operators.

**Comment by anki-code** [comment]
Hey @tomodachi94 ! Try to start here - https://xon.sh/editors.html

**Reply by tomodachi94** [reply]
Appreciated!

**Reply by jnoortheen** [reply, marked as answer]
The vscode syntax is a Python syntax. It still needs some Xonsh specific syntax be added. You could check the [Xonsh's pygments lexer](https://github.com/xonsh/xonsh/blob/be69ff3ad037e8afdc311b0d3c901fdb496326f6/xonsh/pyghooks.py#L1637), which is complete.
